spring/rocketcorner/src/main/java/com/example/rocketcorner/controller/UserRoutes.java

```diff
@@ -75,21 +76,43 @@
 
     @PostMapping("/login")
     public ResponseEntity<?> login(@RequestParam String username, @RequestParam String password) throws ExecutionException, InterruptedException {
-        HashMap<String, User> allUsersHash = firebaseService.getAllUsers();
-        for( Map.Entry<String, User> entry : allUsersHash.entrySet()){
-            if(entry.getValue().getUsername().equals(username)){
-                if(entry.getValue().getPassword().equals(password)){
+        try {
+            HashMap<String, User> allUsersHash = firebaseService.getAllUsers();
+            for (Map.Entry<String, User> entry : allUsersHash.entrySet()) {
+                if (entry.getValue().getUsername().equals(username)) {
+                    if (entry.getValue().getPassword().equals(password)) {
 
-                    return new ResponseEntity<>(entry.getKey(), HttpStatus.OK);
+                        return new ResponseEntity<>(entry.getKey(), HttpStatus.OK);
+                    }
                 }
             }
+            return new ResponseEntity<>("Invalid Login Credentials", HttpStatus.FORBIDDEN);
+        } catch (Exception e){
+            System.out.print(e);
+            return new ResponseEntity<>("INTERNAL SERVER ERROR", HttpStatus.INTERNAL_SERVER_ERROR);
         }
-        return new ResponseEntity<>("Invalid Login Credentials", HttpStatus.FORBIDDEN);
     }
 
     @DeleteMapping("/deleteUser")
-    public ResponseEntity<?> deleteUser(@RequestParam String userId, @RequestParam String password)  {
-        return new ResponseEntity<>(userId + " Deleted", HttpStatus.OK);
+    public ResponseEntity<?> deleteUser(@RequestParam String userId, @RequestParam String password) throws ExecutionException, InterruptedException {
+        try {
+            HashMap<String, User> userHash = firebaseService.getUser(userId);
+            if (userHash != null){
+                for(Map.Entry<String, User> entry : userHash.entrySet()){
+                    if(entry.getValue().getPassword().equals(password) || password.equals(Admin.ADMIN_PASSWORD)){
```

Comment:
(entry.getValue().getPassword().equals(password) && entry.getKey().equals(userId)) || (password.equals(Admin.ADMIN_PASSWORD) && password.equals(Admin.ADMIN_USERNAME))

is necessary because 2 users could have the same password